### DISCUSSION TITLE
SOLWEIG: Meteorological data for diffuse and direct-beam shortwave radiation

### DISCUSSION THREAD

**Discussion by jlegewie** [UMEP-dev/UMEP#577] [Q&A] [answered]
Hi,

we would like to improve the performance of our SOLWEIG model by incorporating diffuse and direct beam solar radiation, instead of estimating them based on Reindl et al. (1990) (`ONLYGLOBAL` setting).

However, we are unsure about the correct measures. The documentation refers to "Diffuse shortwave radiation" and "Direct shortwave radiation" and mentions that "direct radiation perpendicular to the solar beam should be used".

ERA5 does not include the parameters directly. We are now looking at data from the National Solar Radiation Database ([NSRDB](https://nsrdb.nrel.gov)). NSRDB includes measures for global horizontal irradiance (GHI), direct normal irradiance (DNI) or beam radiation and diffuse horizontal irradiance (DHI) (as well as clearsky version of each). Here are our questions:

1. Are these appropriate measures? 
2. Do we need to calculate direct-beam from global and diffuse radiation (instead of using DNI) using the formula listed [here](https://umep-docs.readthedocs.io/en/latest/OtherManuals/SOLWEIG.html#meteorological-data) because it is not perpendicular to the solar beam?

I am including some figures that compare `kdown` from ERA5 to the measures from NSRDB in New York City.

### Daily maximum in global radiation for NYC in 2020 (Monthly average)
<img width="884" alt="Screenshot 2023-12-07 at 11 38 44 AM" src="https://github.com/UMEP-dev/UMEP/assets/1292401/72949a05-ff11-4aa6-9bde-44d97bed9739">

### NYC, July 9, 2020 (max temp 32 C, low cloud cover)
Black line is `kdown` from ERA5
<img width="886" alt="Screenshot 2023-12-07 at 11 42 38 AM" src="https://github.com/UMEP-dev/UMEP/assets/1292401/4e81184c-8b0e-474a-858a-02baf63555e4">

### NYC, April 29, 2020 (max temp 14.4 C, high cloud cover)
Black line is `kdown` from ERA5. Note that overall the correlation between `kdown` and `GHI` is very high (about 0.98)  but there are some days on which `kdown` is substantially higher. 

<img width="882" alt="Screenshot 2023-12-07 at 11 44 57 AM" src="https://github.com/UMEP-dev/UMEP/assets/1292401/30dea0e7-9973-42ce-a418-0c63554722bb">

**Comment by biglimp** [comment, marked as answer]
So to get the direct beam solar radiation you need to recalculate your horisontal direct beam to perpendicular to the sun beam. As you have all the components (global and diffuse), you can use **direct-beam radiation = (global−diffuse) / sin(sun altitude)**. There are probably other ways but this is the one I use.

One note on this. At very low solar altitudes this might be problematic as you divide with a very small number. Depending on the quality of the data, I usually set up a rule that direct radiation cannot be higher that e.g. 1400 or something.

Another note. The are many ways to calculate sun altitude. One way is to run solweig with a POI included. Then you will get the output sun position for each time step (middle of time step)

Hope this helps.
/Fredrik

**Comment by jlegewie** [comment]
Thanks!

I assume this uses sun altitude in radians, not degrees?

Using the variables in `solweig_algorithm.py` with `altitude` calculated by `Solweig_2015a_metdata_noload`, my code looks like this:

```
altitude_radians = altitude[0] * (math.pi / 180)
radI = np.where(altitude[0] > 0, (radG - radD) / np.sin(altitude_radians), 0)
radI = 1400 if radI > 1400 else radI
```

I am also setting `radI` to 0 if altitude <= 0.

**Reply by biglimp** [reply]
Yes, in radians.
